linux_pipeline/Jenkinsfile_lis_rpm_hyperv

```diff
@@ -0,0 +1,177 @@
+               "ws2012r2":"rhel_6.4_32bit,rhel_6.4_x64,rhel_6.5,rhel_6.6_x64,rhel_6.6_x32, \
+                    centos_6.4_x64,centos_6.5_x64,", \
+               "ws2016":"oracle_6.5_rhck,oracle_6.9_rhck, \
+                    oracle_7.4_rhck,oracle_7.0_rhck,centos_7.0_x64,centos_7.0_gen2vm, \
+                    centos_7.2,rhel_7.0,rhel_7.1,rhel_7.2_ga,rhel_7.3_gen2vm,", \
+                "sriov":"rhel_7.3,rhel_7.4,centos_7.4,centos_7.4_x64,centos_7.3,"]
+                    
+def supportedDistros = nodesMap["ws2012"] + nodesMap["ws2012r2"] + nodesMap["ws2016"] + nodesMap["sriov"]
+def deployNodes = "sriov"
+
+def PowerShellWrapper(psCmd) {
+    psCmd = psCmd.replaceAll("\r", "").replaceAll("\n", "")
+    bat "powershell.exe -NonInteractive -ExecutionPolicy Bypass -Command \"\$ErrorActionPreference='Stop';[Console]::OutputEncoding=[System.Text.Encoding]::UTF8;$psCmd;EXIT \$global:LastExitCode\""
+}       
+
+def getDistros(distros,supportedDistros){
+    def validatedDistros = []
+    suppList = supportedDistros.split(",")
+    distros = distros.split(",")
+    
+    for (distro in distros){
+        distroType = distro.split("_")[0]
+        distroVersion = distro.split("_")[1]
+        
+        if (distroVersion.toLowerCase() == "all"){
+            for (item in suppList){
+                if (distroType.toLowerCase() == item.split("_")[0]){
+                    validatedDistros << item
+                }
+            }
+        } else {
+            if (supportedDistros.contains(distro.toLowerCase())){
+                validatedDistros << distro.toLowerCase()
+            }
+        }
+    }
+    return validatedDistros
+}
+
+stage("Deploy stage"){
+    def runs = [:]
+    DISTROS = getDistros(DISTRO_VERSIONS, supportedDistros)
+    nodesMap.keySet().each{
+        node = it
+        mappedDistros = nodesMap[it]
+        if (node != 'sriov'){
+            node = 'ws2016'
+        }
+        DISTROS.each{
+            if (mappedDistros.contains("${it},")){
+                runs["${it}-${node}"] = {
+                    node("${node}"){
+                        stage("${it}-${node}"){
+                            withCredentials(bindings: [string(credentialsId: 'AZURE_SAS', variable: 'AZURE_SAS'),
+                                        string(credentialsId: 'LISA_IMAGES_SHARE_URL', variable: 'LISA_IMAGES_SHARE_URL'),
+                                        string(credentialsId: 'LISA_TEST_DEPENDENCIES', variable: 'LISA_TEST_DEPENDENCIES')]) { 
+                                LIS_SUCCESS = false
+                                if (node == 'sriov'){
+                                    lisSuite = 'lis_deploy_scenarios_sriov'
+                                } else {
+                                    lisSuite = 'lis_deploy_scenarios'
+                                }
+                                dir ("d${BUILD_NUMBER}"){
+                                    checkout scm
```

Comment:
why do we need the checkout scm over here?

Reply:
by dafault the rebo wont be cloned , the pipeline will just run the jenkinsfile